PoSSapi/src/WebUI/Controllers/ReservationController.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.AspNetCore.Mvc;
+using PoSSapi.Application.Common.Models;
+using PoSSapi.Application.Reservations;
+using PoSSapi.Application.Reservations.Commands;
+using PoSSapi.Domain.Entities;
+using PoSSapi.Domain.Enums;
+
+namespace PoSSapi.WebUI.Controllers;
+
+public class ReservationController : ApiControllerBase
+{
+    [HttpGet("{id}")]
+    public async Task<ActionResult<ReservationDto>> GetReservationById(Guid id)
+    {
+        return await Mediator.Send(new GetReservationByIdQuery(id));
+    }
+
+    [HttpGet("list")]
+    public async Task<ActionResult<PaginatedList<ReservationDto>>> GetAllReservations([FromQuery] GetAllReservationsQuery query)
+    {
+        return Ok(await Mediator.Send(query));
+    }
+
+    [HttpGet("list/{userId}")]
+    public async Task<ActionResult<IQueryable<ReservationDto>>> GetUsersReservations(Guid userId)
+    {
+        return Ok(await Mediator.Send(new GetUserReservationsQuery(userId)));
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IQueryable<ReservationDto>>> GetReservationsByStatus([FromQuery] GetReservationsByStatusQuery query)
+    {
+        return Ok(await Mediator.Send(query));
+    }
```

Comment:
`GetAllReservations` and `GetReservationsByStatus`  seem like they could be just one endpoint with status parameter as query. `GetUsersReservation` also seems like it could be just a part of a query instead of a separate endpoint